.github/workflows/freertos_plus_demos.yml

```diff
@@ -380,11 +381,30 @@ jobs:
             FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP
 
       - name: Install prerequisite packages
-        run: sudo apt-get install -y git build-essential libpcap-dev
+        run: sudo apt-get install -y git build-essential libglib2.0-dev libslirp-dev
 
-      - name: Build FreeRTOS+TCP Minimal Demo
+      - name: Start localhost Echo server
+        id: echo-server
+        uses: FreeRTOS/CI-CD-GitHub-Actions/localhost-echo-server@main
+        with:
+          port_number: 5000
+
+      - name: Build FreeRTOS+TCP Echo Posix Demo
+        id: build-echo-posix-demo
         working-directory: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix
-        run: make -j
+        run: |
+          sed -i -z "s/define[[:space:]]*echoECHO_PORT[[:space:]]*([[:space:]]*[0-9]\+[[:space:]]*)/define echoECHO_PORT ( 5000 )/g" TCPEchoClient_SingleTasks.c
+          sed -i -z "s/int[[:space:]]*main[[:space:]]*([[:space:]]*void[[:space:]]*)\n{/int main( void ){setvbuf( stdout, NULL, _IONBF, 0 );/g" main.c
```

Comment:
Non-blocking: Why do these text substitutions from the workflow yaml? Is this so the `main.c` and `TCPEchoClient_SingleTasks.c` files can be reused across multiple demos?

Reply:
The first sed command is to redefine echo port to be 5000, in order to connect to the localhost echo server setup as part of the git workflow. Default port is 7.
The second sed command is to force stdout to write immediately and display the output while git workflow is running.

Reply:
Thanks for the explanation. I was able to understand _what_ these commands do but not why you need them. Why do we need to override the port and output settings rather than simply changing them in code?

Reply:
Adding this command to ensure the demo in git workflow is always built and connect to port 5000, regardless what port number is in the code. Just in case the port number in the code is changed to something else.